### PR TITLE
Throwing while spinning or flipping increases its throw power, but tires you out

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Throwing while spinning or flipping increases its throw power. However, it also applies 35 stamina damage.

## Why It's Good For The Game

It's funny to spin while throwing and now there's an actual reason to do it. Increasing throwing power is noticeably impactful, though, so it causes some stamina damage to deter spam and give a drawback - if you accidentally throw while spipping once it won't be real cause for concern.

## Changelog

:cl:
add: Throwing while spinning or flipping increases its throw power, but tires you out
/:cl:

### CODE REVIEW

**Comment by deathrobotpunch** [comment]
what about rotatium?

File: code/__HELPERS/visual_effects.dm [hunk 1]
```diff
@@ -80,6 +80,10 @@
  * * parallel: whether the animation calls have the ANIMATION_PARALLEL flag, necessary for it to run alongside concurrent animations.
  */
 /atom/proc/SpinAnimation(speed = 1 SECONDS, loops = -1, clockwise = TRUE, segments = 3, parallel = TRUE)
+	// this is most commonly used with the 'flip' emote, so we are just calling it that
+	if(loops != INFINITE)
+		ADD_TRAIT(src, TRAIT_FLIPPING, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(src, TRAIT_FLIPPING, REF(src)), speed * loops)
```
Comment (MrMelbert) [inline code/__HELPERS/visual_effects.dm:86]: Why not just put it in the flip emote? You risk contaminating a lot of instances that probably don't intend on having this side effect

**Comment by SomeoneYouProbablyKnow** [comment]
This is funny but it's another semi secret combo technique for extra effects, do we want more unnecessary complexity

**Comment by carlarctg** [comment]
> This is funny but it's another semi secret combo technique for extra effects, do we want more unnecessary complexity

yes

**Comment by Autisem** [comment]
should this also not decrease accuracy? I cant imagine you could accuratly throw a spear while spinning

**Comment by mc-oofert** [comment]
> should this also not decrease accuracy? I cant imagine you could accuratly throw a spear while spinning

people do this for sport

File: code/__HELPERS/visual_effects.dm (hunk 1 above)
Comment (carlarctg) [reply code/__HELPERS/visual_effects.dm:86]: i think thats a feature but okk

File: code/__HELPERS/visual_effects.dm [hunk 2]
```diff
@@ -44,9 +44,9 @@
 	speed /= segments
 
 	if(parallel)
-		animate(src, transform = matrices[1], time = speed, loop = loops, flags = ANIMATION_PARALLEL)
+		animate(src, transform = matrices[1], time = speed, loops , flags = ANIMATION_PARALLEL)
 	else
-		animate(src, transform = matrices[1], time = speed, loop = loops)
+		animate(src, transform = matrices[1], time = speed, loops)
```
Comment (MrMelbert) [inline code/__HELPERS/visual_effects.dm:49]: Why'd we remove these?

Comment (carlarctg) [reply code/__HELPERS/visual_effects.dm:49]: probably rebase gunk

**Comment by Loafin34** [comment]
could you add a tip of the round as documentation?

**Comment by zoomachina** [comment]
image

**Comment by Eronymun** [comment]
Does this stack with the throwing arm quirk?

**Comment by Time-Green** [comment]
I’m really not fond of adding more secret combat techs and also legitimizing people spamming the spin keybind even more

Find a maintainer who is onboard with this, otherwise I’ll close it in a day (also sorry for not doing this earlier and blueballing you now)

**Pull request closed without merging**